### PR TITLE
feat(windows): Implemented gethost() for windows

### LINKED ISSUE #1549: GetDeviceName() only returns only 15 characters in windows machine.
## Summary

|             |     |
| ----------- | --- |
| Version     | ?   |
| Affected OS | Windows  |
| OS Version  | ?   |

## Current behavior

GetDeviceName() only returns only 15 characters in windows machine.

## Expected behavior

In windows computer name is 63 characters and we need whole name to be returned. Probably need to implement getHost() for windows.

### PR DESCRIPTION
**Pull request by sohail-p**
## Description
Added windows implementation of getHost(). Using this user can get full name of windows devices.
Fixes #1549

## Compatibility

| OS      | Existing| Implemented |
| ------- | :---------: | :---------: |
| iOS     |    ❌     |    ❌     |
| Android |    ✅     |    ✅     |
| Windows |    ❌     |    ✅     |

## Checklist

* [x] I have tested this on a device/simulator for each compatible OS
* [x] I added the documentation in `README.md`
* [] I updated the typings files (`privateTypes.ts`, `types.ts`)
* [] I added a sample use of the API (`example/App.js`)

### CODE REVIEW

**Comment by sohail-p** [comment]
@mikehardy can you please merge this at your convenience?

**Comment by lone-w** [comment]
@mikehardy  Can you please merge this today.

File: windows/code/RNDeviceInfoCPP.h [hunk 1]
```diff
@@ -833,6 +837,25 @@ namespace winrt::RNDeviceInfoCPP
           promise.Resolve(op.GetResults());
         });
     }
+	
+	REACT_SYNC_METHOD(getHostSync);
+    std::string getHostSync() noexcept
```
Comment (mikehardy) [inline windows/code/RNDeviceInfoCPP.h:842]: Spaces vs tabs - no auto-formatter or check on spacing on windows I guess? Trivial change though, will commit it and squash-merge to push this through

```suggestion
    REACT_SYNC_METHOD(getHostSync);
    std::string getHostSync() noexcept
```

**Review by mikehardy — APPROVED** [review]
Looks good - final QA/CI checks then can merge
The repeated pings are borderline on getting a comment for "please be patient", remember open source maintainers are your collaborators, not your service providers and timing expectations can feel burdensome

However! Just taking the time to post a PR and help everyone that uses the package is awesome, and I sincerely appreciate that. With my limited connectivity during travel right now the limited ability to find work threads etc means the repeated pings actually surfaced the helpful links to move this forward in easily-clickable way so it balances out ;-)

Hopefully released in a couple minutes

Cheers

**Comment by sohail-p** [comment]
E2E was passing earlier, now it failed for `java.lang.RuntimeException: android.os.DeadSystemException`
image

**Comment by mikehardy** [comment]
> E2E was passing earlier, now it failed for `java.lang.RuntimeException: android.os.DeadSystemException` image

I need to change the way the android CI emulator is started up, I manage quite a few repos and don't have CI working in the most optimal way in all of them all at the same time (always discovering new/better methods, but takes time to propagate...) in the meantime the emulator startup here is flaky. Eventually will have it handled same way I'm doing either react-native-firebase or ankidroid  as those seem to be working well at the moment

**Pull request merged**